crates/fuel-core/src/service/adapters/compression_adapters.rs

```diff
@@ -60,20 +63,25 @@ impl block_source::BlockSource for CompressionBlockImporterAdapter {
         self.block_importer.events_shared_result()
     }
 
-    fn get_block(&self, height: BlockAt) -> anyhow::Result<SharedImportResult> {
-        self.import_result_provider_adapter
-            .result_at_height(height.into())
+    fn get_block(&self, height: BlockAt) -> anyhow::Result<Block> {
+        let block = &self
+            .import_result_provider_adapter
+            .result_at_height(height.into())?
+            .sealed_block
+            .entity;
+        Ok(block.clone())
```

Comment:
do we need this clone here? curious 

perhaps we can clone the `SharedImportResult` first and then return the entity from that because it's wrapped in an Arc. maybe wrong here

Reply:
`import_result_provider_adapter` uses state rewind under the hood=DDD

We need just to get the block from the database, not from the `import_result_provider_adapter`

Reply:
Changed it to `OnChain` db, so its not an issue now.